packages/react/src/components/CodeSnippet/__tests__/CodeSnippet-test.js

```diff
@@ -6,102 +6,188 @@
-    });
+  it('should render children as expected', () => {
+    render(
+      <CodeSnippet type="inline" data-testid="code-4">
+        {inline}
+      </CodeSnippet>
+    );
+
+    expect(screen.getByTestId('code-4')).toHaveTextContent(inline);
   });
 
-  describe('check for showMoreBtn', () => {
-    let wrapper;
+  it('should allow custom classes to be applied when passed in via className', () => {
+    render(
+      <CodeSnippet type="inline" data-testid="code-5" className="custom-class">
+        {inline}
+      </CodeSnippet>
+    );
 
-    beforeEach(() => {
-      wrapper = mount(<CodeSnippet type={'multi'} />);
-    });
+    expect(screen.getByTestId('code-5')).toHaveClass('custom-class');
+  });
 
-    it('when less then 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14',
-      });
+  it('should allow hiding the copy button', () => {
+    render(
+      <CodeSnippet type="single" hideCopyButton>
+        {single}
+      </CodeSnippet>
+    );
 
-      expect(wrapper.find(Button).length).toBe(0);
-    });
+    expect(document.querySelector('button')).not.toBeInTheDocument();
+  });
 
-    it('when exactly 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15',
-      });
+  it('should set disabled on copy button if it is passed via props', () => {
+    render(
+      <CodeSnippet type="single" disabled>
+        {single}
+      </CodeSnippet>
+    );
 
-      expect(wrapper.find(Button).length).toBe(0);
-    });
+    expect(screen.getByTitle('Copy to clipboard')).toHaveAttribute('disabled');
+  });
+});
+
+describe('CodeSnippet events', () => {
+  it('should call the click handler when the copy button is clicked', () => {
+    const onClick = jest.fn();
+    render(
+      <CodeSnippet type="single" onClick={onClick}>
+        {single}
+      </CodeSnippet>
+    );
+
+    const button = screen.getByTitle('Copy to clipboard');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it('should call the click handler with type inline', () => {
+    const onClick = jest.fn();
+    render(
+      <CodeSnippet type="inline" data-testid="code-6" onClick={onClick}>
+        {inline}
+      </CodeSnippet>
+    );
+
+    const button = screen.getByTestId('code-6');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+});
+
+describe('Show more button', () => {
+  it('should not have show more button when less then 15 rows', () => {
+    render(<CodeSnippet type="multi">{multiShort}</CodeSnippet>);
+
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
+    ).not.toBeInTheDocument();
+  });
+
+  it('should not have show more button when exactly 15 rows', () => {
+    render(<CodeSnippet type="multi">{multi15}</CodeSnippet>);
+
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
+    ).not.toBeInTheDocument();
```

Comment:
Would it be possible to query for this without `querySelector`? Maybe `screen.queryByText()` and use the text for "show more" and assert that it is null?

Reply:
@joshblack I tried this but was running into errors bc it's not in the document at all, so it kept failing with that selector
https://github.com/carbon-design-system/carbon/pull/11665#discussion_r904173752

Reply:
ahh thank you! fixed it 👍🏽